### PR TITLE
Support more sophisticated proxy settings

### PR DESCRIPTION
**Pull request by mattesh**
The commit https://github.com/notepad-plus-plus/wingup/commit/17f2ee0a3f85f48ccad7d5fecc02d4b59f21b262
from gwinkless was giving me the right direction, but was not sufficient for my Kerberos based proxy system.
I added required parameters to treat curl as required for different constellations, trying to do as minimal changes to the rest.
@donho and @gwinkless could you check from your end if this changes are OK to release?
Regards, Mattes

### CODE REVIEW

**Comment by Neustradamus** [comment]
@donho: What do you think?

**Comment by mattesh** [comment]
@donho,
Is there anything blocking this to be merged?
We did tests and see this as crucial change for new releases of NPP+ Updater
Please, let me know if something should be added.
Thank you
/BR Mattes

File: src/xmlTools.h [hunk 1]
```diff
@@ -75,12 +75,20 @@ class GupExtraOptions : public XMLTool {
...
-	void writeProxyInfo(const wchar_t *fn, const wchar_t *proxySrv, long port);
+	void setProxyServer(const std::wstring& s) { _proxyServer = s; };
+	long getPort() const { return _port; };
+	void setPort(long l) { _port = l; };
+	bool hasProxySettings() const {return ((!_proxyServer.empty()) && (_port != -1)); };
+	void write();
+	const std::wstring& getAuth() const { return _proxyAuth; };
+	void setAuth(const std::wstring& s) { _proxyAuth = s; };
+	const std::wstring& getUserPass() const { return _proxyUserPass; };
+	void setUserPass(const std::wstring& s) { _proxyUserPass = s; };
```
Comment (donho) [inline src/xmlTools.h:86]: User password shouldn't be stored into the xml file.

File: vcproj/GUP.vcxproj [hunk 2]
```diff
@@ -158,7 +158,7 @@
     </ClCompile>
     <Link>
       <AdditionalDependencies>libcurl_debug.lib;ZipLib.lib;zlib.lib;lzma.lib;bzip2.lib;comctl32.lib;shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
-      <AdditionalLibraryDirectories>..\curl\builds\libcurl-vc15-x64-debug-dll-ipv6-sspi-schannel\lib;Bin\x64\Debug;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(ProjectDir)..\curl\builds\libcurl-vc15-$(LibrariesArchitecture)-$(Configuration)-dll-ipv6-sspi-schannel\lib;Bin\x64\Debug;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (donho) [inline vcproj/GUP.vcxproj:161]: Why does this file and the 2 following files are modify?

File: vcproj/GUP.vcxproj [hunk 3]
```diff
@@ -146,7 +146,7 @@
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <ClCompile>
       <Optimization>Disabled</Optimization>
-      <AdditionalIncludeDirectories>..\src\TinyXml;..\src\sha2;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (donho) [inline vcproj/GUP.vcxproj:149]: Why is this file modified?

File: src/winmain.cpp [hunk 4]
```diff
@@ -930,13 +992,19 @@ bool getUpdateInfo(const string& info2get, const GupParameters& gupParams, const
...
-			curl_easy_setopt(curl, CURLOPT_PROXY, ws2s(proxyServer.getProxyServer()).c_str());
-			curl_easy_setopt(curl, CURLOPT_PROXYPORT, proxyServer.getPort());
-			curl_easy_setopt(curl, CURLOPT_PROXYAUTH, CURLAUTH_ANY);
-		}
+			if (!extraOptions.getAuth().empty()) {
+				long lAuth = ProxyAuthType::getAuthNum(extraOptions.getAuth());
 
+				curl_easy_setopt(curl, CURLOPT_PROXYAUTH, lAuth); // any method as proposed by proxy
+			}
+			if (!extraOptions.getUserPass().empty()) {
```
Comment (donho) [inline src/winmain.cpp:1002]: Coding convention is not respected.

**Comment by donho** [comment]
The KISS (Keep it Simple, Stupid) guideline should be respected.
However, this PR is complex than it should be, also there are so many stuff added into UI which makes UX difficult.
Therrefore the PR is not accepted.
